watcher/watcher.go

```diff
@@ -0,0 +1,156 @@
+package watcher
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Tonkpils/snag/exchange"
+	fsn "gopkg.in/fsnotify.v1"
+)
+
+const rebuild = "rebuild"
+
+type Watcher interface {
+	Watch(string) error
+}
+
+type FSWatcher struct {
+	ex           exchange.SendListener
+	fsn          *fsn.Watcher
+	done         chan struct{}
+	mtimes       map[string]time.Time
+	watching     map[string]struct{}
+	watchDir     string
+	ignoredItems []string
+}
+
+func New(ex exchange.SendListener, ignoredItems []string) (Watcher, error) {
```

Comment:
Regarding the side note we could break up the interface up a little bit:

```go
type Sender interface {
    Send(string, interface{})
}

type Listener interface {
    Listen(string, func(interface{})
}

type SendListener interface {
    Sender
    Listener
}
```

and then pass a `Sender` to the watcher.